radixdlt/src/main/java/com/radixdlt/consensus/VertexMetadata.java

```diff
@@ -42,27 +42,42 @@
 	@DsonOutput(Output.ALL)
 	private final Hash id;
 
+	@JsonProperty("stateVersion")
+	@DsonOutput(Output.ALL)
+	private final long stateVersion;
+
 	VertexMetadata() {
 		// Serializer only
 		this.view = null;
 		this.id = null;
+		this.stateVersion = 0L;
 	}
 
-	public VertexMetadata(View view, Hash id) {
+	public VertexMetadata(View view, Hash id, long stateVersion) {
+		if (stateVersion < 0) {

```

Comment:
If it is an invariant that stateVersion >= 0, then I would suggest adding custom serialization methods (same as view) that actually make this check to ensure that deserialized objects respect this invariant.

Reply:
It is indeed an invariant but don't think it's a serialization level invariant. Proof of malfeasance would be useful for slashing later.

Reply:
Ah, OK.  It gets hashed in the VoteData, so should be fine